subprojects/core/src/test/groovy/org/gradle/api/internal/tasks/options/OptionReaderTest.groovy

```diff
@@ -169,6 +185,15 @@ class OptionReaderTest extends Specification {
         e.message == "Option 'stringValue' on method cannot take multiple parameters in class 'org.gradle.api.internal.tasks.options.OptionReaderTest\$TestClass4#setStrings'."
     }
 
+    def "fails when option name starts with 'no-'"() {
+        when:
+        reader.getOptions(new TestClass6());
+        then:
+        def e = thrown(OptionValidationException)
+        println(e.message)
+        e.message == "Illegal option name 'no-optionName' in class 'org.gradle.api.internal.tasks.options.OptionReaderTest\$TestClass6'. Option name must not start with 'no-'."
+    }
```

Comment:
:x: This is a breaking change for builds that have task which declare a boolean option whose name starts with `no-`.

What we should do instead is only register disabling command line options iff they don't conflict with a declared option. See `OptionDescriptor.isClashing()`.